report/sections/Frontend/client-server-interaction.tex

```diff
@@ -0,0 +1,46 @@
+\section{Client-Server Interaction}
+When a lecturer creates a new exercise, a modal prompts them to enter details about the exercise.
+These details include the name of the exercise, instructions for the student, an optional code template to help the student get started, as well as test code.
+This data is sent to the database where it is stored.
+When a student opens an exercise, the data is then fetched from the database.
+This is done by the client sending a request to the Next.js backend, which queries the database and returns both exercise and test data.
+
+\begin{figure}[H]
+    \centering
+    \includegraphics[scale=0.1]{create_exercise.png}
+    \caption{Creating a new exercise as a lecturer}
+    \label{fig:create_exercise}
+\end{figure}
+
+In the interest of reducing the number of database queries, the result of this request is cached on the client.
+The client will eagerly request new data to stay up to date with instructions or test code updates.
+We deemed that this short cache storage time is useful for when lecturers update test code during an exercise session.
+This way, all students have the latest test code shortly after the update while still minimizing unnecessary queries.
+
+When a user submits their exercise solution attempt, a mutation is sent to the Next.js backend.
+In this request, the necessary code and test code is included.
+One could argue that fetching the test code on the backend when processing this mutation would be more appropriate.
+However, we want to eventually display the test code to the user so they may gain a better understanding of the solution requirements, although this will not be done in this iteration of the project.
```

Comment:
```suggestion
```

Reply:
If this is removed, the previous sentence should be modified